packages/camera/camera/example/lib/main.dart

```diff
@@ -468,49 +479,52 @@ class _CameraExampleHomeState extends State<CameraExampleHome>
 
   /// Display the control bar with buttons to take pictures and record videos.
   Widget _captureControlRowWidget() {
+    VoidCallback? takePictureButton;
+    VoidCallback? recordButtonPressed;
+    VoidCallback? pauseOrResumeButtonPressed;
+    VoidCallback? stopButtonPressed;
+
+    if (controller != null) {
+      final CameraController cameraController = controller!;
```

Comment:
Here as well, you can do this first (as `CameraController?`), use it in the condition, and then everything else will automatically know that it's non-null.

Reply:
The "(as `CameraController?`)" was the magic hint I needed. Now I understand that the analyser can figure out that a null check is done when dealing with local variables. 

Thanks for this feedback!